### PR TITLE
Statistics: release info

### PR DESCRIPTION
**Pull request by hedgehog1833**
Almost through with the statistics on backend side now. Only ImportInfo missing after this.

### CODE REVIEW

File: butler/src/test/java/rocks/metaldetector/butler/client/transformer/ButlerStatisticsTransformerTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    var releaseInfo = ReleaseInfoDto.builder().build();
+    var butlerStatistics = ButlerStatisticsDto.builder().releaseInfo(releaseInfo).build();
+    doReturn(butlerStatistics).when(objectMapper).convertValue(any(), eq(ButlerStatisticsDto.class));
+
+    // when
+    ButlerStatisticsDto result = underTest.transform(ButlerStatisticsResponse.builder().build());
+
+    // then
+    assertThat(result).isEqualTo(butlerStatistics);
+  }
```
Comment (DanielW1987) [inline butler/src/test/java/rocks/metaldetector/butler/client/transformer/ButlerStatisticsTransformerTest.java:64]: If the transformation failed due to incompatible naming, no test will fail.

File: butler/src/test/java/rocks/metaldetector/butler/client/ReleaseButlerRestClientImplTest.java [hunk 2]
```diff
@@ -557,4 +559,75 @@ private Stream<Arguments> httpStatusCodeProvider() {
       return Stream.of(HttpStatus.values()).filter(status -> !status.is2xxSuccessful()).map(Arguments::of);
     }
   }
+
+  @DisplayName("Tests getting statistics")
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
```
Comment (DanielW1987) [inline butler/src/test/java/rocks/metaldetector/butler/client/ReleaseButlerRestClientImplTest.java:564]: You can omit that...

```suggestion
```

File: butler/src/test/java/rocks/metaldetector/butler/client/ReleaseButlerRestClientImplTest.java [hunk 3]
```diff
@@ -557,4 +559,75 @@ private Stream<Arguments> httpStatusCodeProvider() {
...
+      doReturn(ResponseEntity.status(httpStatus).body(ButlerStatisticsResponse.builder().build())).when(restTemplate).getForEntity(anyString(), eq(ButlerStatisticsResponse.class));
+
+      // when
+      Throwable throwable = catchThrowable(() -> underTest.getStatistics());
+
+      // then
+      assertThat(throwable).isInstanceOf(ExternalServiceException.class);
+    }
+
+    private Stream<Arguments> httpStatusCodeProvider() {
```
Comment (DanielW1987) [inline butler/src/test/java/rocks/metaldetector/butler/client/ReleaseButlerRestClientImplTest.java:629]: and now use private static classes in inner classes. The TestInstance workaround is no longer necessary.

```suggestion
    private static Stream<Arguments> httpStatusCodeProvider() {
```

File: webapp/src/main/java/rocks/metaldetector/service/statistics/StatisticsServiceImpl.java [hunk 4]
```diff
@@ -68,4 +69,14 @@ private ArtistFollowingInfo buildArtistFollowingInfo() {
...
+
+  private ReleaseInfo getReleaseInfo(ReleaseInfoDto releaseInfoDto) {
+    return objectMapper.convertValue(releaseInfoDto, ReleaseInfo.class);
+  }
+
+  private List<ImportInfo> getImportInfo(List<ImportInfoDto> importInfoDtos) {
+    return importInfoDtos.stream()
+        .map((importInfo) -> objectMapper.convertValue(importInfo, ImportInfo.class))
+        .collect(Collectors.toList());
+  }
```
Comment (DanielW1987) [inline webapp/src/main/java/rocks/metaldetector/service/statistics/StatisticsServiceImpl.java:81]: tbh i'm not a big fan of the objectmapper approach. For me, it's ok to use the `ImportInfoDto` throughout the app, like we did with `SpotifyArtistDto`.

**Review by DanielW1987 — APPROVED** [review]

File: webapp/src/main/java/rocks/metaldetector/service/statistics/StatisticsServiceImpl.java (hunk 4 above)
Comment (hedgehog1833) [reply webapp/src/main/java/rocks/metaldetector/service/statistics/StatisticsServiceImpl.java:81]: Yeah sounds good. I went the lazy way 😉 but the part with the tests convinces me..

**Pull request merged**